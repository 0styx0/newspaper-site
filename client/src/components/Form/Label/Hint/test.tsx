import * as React from 'react';
import * as renderer from 'react-test-renderer';
import HintContainer from './container';
import { mount } from 'enzyme';
import casual from '../../../../tests/casual.data';

describe('<HintContainer>', () => {

    describe('snapshots', () => {

        /**
         * Takes snapshots
         */
        function snap(title: string) {

            const tree = renderer.create(
                <HintContainer
                    title={title}
                    children={<label>Hi<input type="text" value="hello" /></label>}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();

        }

        it('renders correctly', () => snap('Hello'));
    });

    function setup() {

        return mount(
            <HintContainer
              title={casual.sentences(casual.randomPositive)}
              children={<label>Hi<input type="text" /></label>}
            />
        );
    }

    /**
     * Shows hint
     */
    function reveal(wrapper: any) {

        wrapper.find('abbr').first().simulate('click');

        expect((wrapper.find('.abbrMessage').first() as any).nodes).toHaveLength(1);
    }

    describe('hint is revealed when', () => {

        it('is clicked', () => {

            const wrapper = setup();

            wrapper.find('abbr').first().simulate('click');

            expect((wrapper.find('.abbrMessage').first() as any).nodes).toHaveLength(1);
        });

        test(`its props.children is invalid (after user has done some input)`, () => {

            const wrapper = setup();

            wrapper.find('input').first().simulate('input', {target: {checkValidity: () => false}});

            expect((wrapper.find('.abbrMessage').first() as any).nodes).toHaveLength(1);
        });
    });

    describe('hint is hidden when', () => {

        test('component is created', () => {

            const wrapper = setup();

            expect((wrapper.find('.abbrMessage').first() as any).nodes).toHaveLength(0);
        });

        it('is clicked a second time', () => {

            const wrapper = setup();
            reveal(wrapper);

            wrapper.find('abbr').first().simulate('click');

            expect((wrapper.find('.abbrMessage').first() as any).nodes).toHaveLength(0);
        });

        test(`its input becomes valid (after user has done some input)`, () => {

            const wrapper = setup();
            reveal(wrapper);

            wrapper.find('input').first().simulate('input', {target: {checkValidity: () => true}});

            expect((wrapper.find('.abbrMessage').first() as any).nodes).toHaveLength(0);
        });
    });
});
