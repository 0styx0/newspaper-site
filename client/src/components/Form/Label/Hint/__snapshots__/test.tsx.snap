// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HintContainer> snapshots renders correctly 1`] = `
<label
  onInput={[Function]}
>
  Hi
  <input
    type="text"
    value="hello"
  />
  <span>
    <abbr
      onClick={[Function]}
      title="Hello"
    >
      ?
    </abbr>
    <span />
  </span>
</label>
`;
